pkg/component/networking/istio/logging.go

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package istio
+
+import (
+	fluentbitv1alpha2 "github.com/fluent/fluent-operator/v3/apis/fluentbit/v1alpha2"
+	fluentbitv1alpha2filter "github.com/fluent/fluent-operator/v3/apis/fluentbit/v1alpha2/plugins/filter"
+	fluentbitv1alpha2parser "github.com/fluent/fluent-operator/v3/apis/fluentbit/v1alpha2/plugins/parser"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/component"
+)
+
+const (
+	istioProxyParserName = "istio-proxy-parser"
+)
+
+// CentralLoggingConfiguration returns a fluent-bit parser and filter for the cluster-autoscaler logs.
+func CentralLoggingConfiguration() (component.CentralLoggingConfig, error) {
+	return component.CentralLoggingConfig{Filters: generateClusterFilters(), Parsers: generateClusterParsers()}, nil
+}
+
+func generateClusterFilters() []*fluentbitv1alpha2.ClusterFilter {
+	return []*fluentbitv1alpha2.ClusterFilter{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   v1beta1constants.DefaultSNIIngressServiceName,
+				Labels: map[string]string{v1beta1constants.LabelKeyCustomLoggingResource: v1beta1constants.LabelValueCustomLoggingResource},
+			},
+			Spec: fluentbitv1alpha2.FilterSpec{
+				Match: "kubernetes.*istio-ingressgateway*istio-proxy*",
+				FilterItems: []fluentbitv1alpha2.FilterItem{
+					{
+						Parser: &fluentbitv1alpha2filter.Parser{
+							KeyName:     "log",
+							Parser:      istioProxyParserName,
+							ReserveData: ptr.To(true),
+							PreserveKey: ptr.To(true),
+						},
+					},
+					{
+						Lua: &fluentbitv1alpha2filter.Lua{
+							Call: "add_kubernetes_namespace_name_to_record",
+							Script: corev1.ConfigMapKeySelector{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: v1beta1constants.ConfigMapNameFluentBitLua,
+								},
+								Key: "add_kubernetes_namespace_name_to_record.lua",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func generateClusterParsers() []*fluentbitv1alpha2.ClusterParser {
+	return []*fluentbitv1alpha2.ClusterParser{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   istioProxyParserName,
+				Labels: map[string]string{v1beta1constants.LabelKeyCustomLoggingResource: v1beta1constants.LabelValueCustomLoggingResource},
+			},
+			Spec: fluentbitv1alpha2.ParserSpec{
+				Regex: &fluentbitv1alpha2parser.Regex{
+					Regex: `^.*\.(?<namespace_name>[a-zA-Z0-9_-]+)\.svc\.cluster\.local.*$`,
```

Comment:
```suggestion
					Regex: `^.*\.(?<namespace_name>shoot--[a-zA-Z0-9_-]+)\.svc\.cluster\.local.*$`,
```
This way the regex is not confused in case there is another `svc.cluster.local` string like `kubernetes.default.svc.cluster.local` in the log message (see description of PR #13569)